rpcs3/rpcs3qt/debugger_frame.cpp

```diff
@@ -583,14 +606,12 @@ u64 debugger_frame::EvaluateExpression(const QString& expression)
 {
 	auto thread = cpu.lock();
 
-	if (!thread) return 0;
-
 	bool ok = false;
 
 	// Parse expression(or at least used to, was nuked to remove the need for QtJsEngine)
 	const QString fixed_expression = QRegExp("^[A-Fa-f0-9]+$").exactMatch(expression) ? "0x" + expression : expression;
 	const u64 res = static_cast<u64>(fixed_expression.toULong(&ok, 16));
-	return ok ? res : thread->get_pc();
+	return ok ? res : (!Emu.IsStopped() && m_rsx ? +m_rsx->ctrl->get : (thread ? thread->get_pc() : 0));
```

Comment:
This is horribly nested ternary...
Just use multiple returns so that one can at least set breakpoints.

Reply:
There's nothing wrong with nested ternary whne you do them right IMO, written it better.